packages/shared/lib/services/sync/data/records.service.ts

```diff
@@ -421,57 +421,66 @@ export async function getAllDataRecords(
             }
         }
 
-        let nextCursor = null;
-
-        const result: RawDataRecordResult[] = await query.select(
+        const rawResults: RawDataRecordResult[] = await query.select(
+            // PostgreSQL stores timestamp with microseconds precision
+            // however, javascript date only supports milliseconds precision
+            // we therefore convert timestamp to string (using to_json()) in order to avoid precision loss
```

Comment:
basically we cannot query pg timestamp column into a js Date without losing the microseconds precision (which then make the cursor logic wrong) so all dates must be json encoded string